lib/widgets/conversations/add_conversation_widget.dart

```diff
@@ -1,122 +1,160 @@
-                    Icons.add_comment,
-                    color: Theme.of(context).highlightColor,
-                  ),
-                ),
-              ),
-              if (conversation != null)
-                Container(
-                  child: RaisedButton(
-                    child: Icon(
-                      Icons.delete_forever,
-                      color: Theme.of(context).errorColor,
-                    ),
-                    onPressed: () => this.deleteConversation(
-                        conversation, conversationData, context),
-                  ),
-                )
-            ],
-          ),
-        ),
-      ),
-    );
-  }
-}
+import 'package:chirp_nets/utils/text.dart';
+import 'package:chirp_nets/widgets/common/delete_alert_dialog.dart';
+import 'package:flutter/material.dart';
+
+import 'package:chirp_nets/models/conversation.dart';
+import 'package:chirp_nets/models/user.dart';
+import 'package:chirp_nets/providers/conversations.dart';
+
+class AddConversationWidget extends StatefulWidget {
+  final Conversations conversationData;
+  final User user;
+  final Conversation conversation;
+
+  AddConversationWidget({this.conversationData, this.user, this.conversation});
+
+  @override
+  AddConversationWidgetState createState() {
+    return AddConversationWidgetState();
+  }
+}
+
+class AddConversationWidgetState extends State<AddConversationWidget> {
+  final _formKey = GlobalKey<FormState>();
+  final Map<String, dynamic> formData = {'name': null, 'networkId': null};
+
+  void addConversation(
+      Conversations conversationData, User user, String name, int networkId, ctx) {
+    if (name.isEmpty || networkId == null) {
+      return;
+    }
+    if (widget.conversation != null) {
+      conversationData.updateConversation(
+        widget.conversation.id,
+        name,
+        networkId,
+      );
+    } else {
+      Conversation conv = Conversation(
+        userId: user.id,
+        name: name,
+        networkId: networkId,
+      );
+      conversationData.addConversation(
+        conv.userId,
+        conv.name,
+        conv.networkId,
+      );
+    }
+    Navigator.pop(ctx);
+  }
+
+  void deleteConversation(conversation, provider, ctx) {
+    showDialog(
+      context: ctx,
+      child: DeleteAlertDialog(
+        title:
+            'Are you sure?\nThis will delete all messages associated with this conversation',
+        id: conversation.id,
+        onDelete: widget.conversationData.deleteConversation,
+      ),
+    ).then((_) {
+      Navigator.of(ctx).pop();
+    });
+  }
+
+  void submitForm(context) {
+    if (_formKey.currentState.validate()) {
+      _formKey.currentState.save();
+      print(formData);
+      addConversation(
+        widget.conversationData,
+        widget.user,
+        formData['name'],
+        int.parse(formData['networkId']),
+        context,
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    /* What does this do? */

```

Comment:
The text controllers basically take the requirement for a state away, they handle the changing and setting of the placeholder. This set the placeholder but since it's a stateful widget now it's not necessary